### ISSUE TITLE
[BUG] Coverage report missing some files

### ISSUE THREAD

**Issue by azamat-sharapov** [cenfun/monocart-reporter#33]
Hi. I'm trying to set up Vue CT coverage based on your example here: https://github.com/cenfun/playwright-ct-vue  
The problem is, almost half of the files are missing from coverage report (5 out of 11). Below is some info.

<details>
<summary>Report screenshots</summary>

<img width="774" alt="image" src="https://github.com/cenfun/monocart-reporter/assets/3023661/287179ac-247e-4653-9cf7-983ab8c0593a">

---

<img width="1012" alt="image" src="https://github.com/cenfun/monocart-reporter/assets/3023661/afcc9e58-22ea-435e-a04b-8ff3e3473f92">

</details>

<details>
<summary>playwright-ct.config.ts</summary>

```typescript
import { resolve } from 'path'
import { defineConfig, devices } from '@playwright/experimental-ct-vue'

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
  testDir: './tests/',

  outputDir: './test-results',

  /* The base directory, relative to the config file, for snapshot files created with toMatchSnapshot and toHaveScreenshot. */
  snapshotDir: './playwright/__snapshots__',
  /* Maximum time one test can run for. */
  timeout: 10 * 1000,
  expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
    timeout: 2000,
  },
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: [
    ['list'],
    [
      'monocart-reporter',
      {
        name: 'My Test Report',
        outputFile: './test-results/index.html',
        coverage: {
          // toIstanbul: true,
          // lcov: true,
          unpackSourceMap: true,
          excludeDistFile: true,
          sourceFilter: (sourceName: string) => {
            return sourceName.search(/src\//) !== -1
          },
          entryFilter: (entry: { url: string }) =>
            !entry.url.includes('tailwind'),
        },
      },
    ],
  ],
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',

    /* Port to use for Playwright component endpoint. */
    ctPort: 3100,

    ctViteConfig: {
      // build: {
      //   sourcemap: true,
      // },

      resolve: {
        alias: {
          '@': resolve(__dirname, './src'),
        },
      },
    },
  },

  /* Configure projects for major browsers */
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'] },
    },
    /*
    {
      name: 'firefox',
      use: { ...devices['Desktop Firefox'] },
    },
    {
      name: 'webkit',
      use: { ...devices['Desktop Safari'] },
    },
    */
  ],
})
```
</details>

<details>
<summary>One of the missing files in report</summary>

### VerticalNavDesktop.vue

```vue
<script setup lang="ts">
import type { VerticalNavItem } from './types'

defineProps<{
  navigationItems: VerticalNavItem[]
}>()
</script>

<template>
  <nav class="flex-col" aria-label="Sidebar">
    <ul role="list" class="-mx-2 space-y-1">
      <li v-for="item in navigationItems" :key="item.title">
        <RouterLink
          :to="item.route"
          :class="[
            item.current
              ? 'bg-gray-50 text-emerald-600'
              : 'text-gray-700 hover:text-emerald-600 hover:bg-gray-50',
            'group flex gap-x-3 rounded-md p-2 text-sm leading-6 font-medium',
          ]"
        >
          <component
            :is="item.icon"
            :class="[
              item.current
                ? 'text-emerald-600'
                : 'text-gray-400 group-hover:text-emerald-600',
              'h-6 w-6 shrink-0',
            ]"
            aria-hidden="true"
          />
          {{ item.title }}
        </RouterLink>
      </li>
    </ul>
  </nav>
</template>
```

### VerticalNavDesktop.spec.ts

```typescript
import { test, expect } from './fixture'
import VerticalNavDesktop from '@/VerticalNavDesktop.vue'

test('VerticalNavDesktop', async ({ mount }) => {
  const component = await mount(VerticalNavDesktop, {
    props: {
      navigationItems: [
        {
          title: 'Test',
          route: { name: 'test' },
          icon: 'icon',
          current: true,
        },
        {
          title: 'Test2',
          route: { name: 'test2' },
          icon: 'icon',
          current: false,
        },
      ],
    },
  })

  expect(await component.getByRole('listitem').count()).toBe(2)

  await expect(
    component.getByRole('listitem').first().getByRole('link')
  ).toHaveClass(/text-emerald/)

  await expect(
    component.getByRole('listitem').first().getByRole('link').locator('icon')
  ).toBeVisible()
})
```
</details>

<details>
<summary>fixture.ts</summary>

```typescript
import { test as ctBase, expect } from '@playwright/experimental-ct-vue'
import { addCoverageReport } from 'monocart-reporter'

const test = ctBase.extend<{ autoTestFixture: string }>({
  autoTestFixture: [
    async ({ page }, use) => {
      // console.log('autoTestFixture setup...');
      // coverage API is chromium only
      if (test.info().project.name === 'chromium') {
        await Promise.all([
          page.coverage.startJSCoverage(),
          // page.coverage.startCSSCoverage(),
        ])
      }

      await use('autoTestFixture')

      // console.log('autoTestFixture teardown...');
      if (test.info().project.name === 'chromium') {
        const [jsCoverage] = await Promise.all([
          page.coverage.stopJSCoverage(),
          // page.coverage.stopCSSCoverage(),
        ])
        const coverageList = [...jsCoverage]
        await addCoverageReport(coverageList, test.info())
      }
    },
    {
      scope: 'test',
      auto: true,
    },
  ],
})

export { test, expect }
```
</details>

**Comment by cenfun** [comment]
I found that two filters used:
```js
sourceFilter: (sourceName: string) => {
            return sourceName.search(/src\//) !== -1
          },
entryFilter: (entry: { url: string }) =>
            !entry.url.includes('tailwind'),
```
Can you debug it and confirm that all files are iterated? or the missing files are filttered by the filters?

**Comment by azamat-sharapov** [comment]
@cenfun I actually already did, those files did not go through these filters. I went deep debugging into monocart-reporter source code last week and found out that files get lost somewhere when sourcemap is generated (if I remember correctly, some utility from `source-map` package wasn't returning this file).

**Comment by azamat-sharapov** [comment]
Also FYI, I put `!entry.url.includes('tailwind')` condition, because this file was included in the report (which I don't need):

### playwright/index.html

```html
...
    <script src="https://cdn.tailwindcss.com"></script>
...
```

**Comment by azamat-sharapov** [comment]
@cenfun thank you. I can't share the repo as it's company's private. But I will try to create small demo as soon as I have some capacity.

**Comment by cenfun** [comment]
Hi @azamat-sharapov 
I refactored the coverage related codes, please try latest version `1.6.35` to see if the issue still exists. thanks

**Comment by azamat-sharapov** [comment]
@cenfun unfortunately the issue still exists.

**Comment by cenfun** [comment]
Ok, it will be no more updates until you can provide a demo that reproduces the issue locally, thanks

**Comment by cenfun** [comment]
Hey @azamat-sharapov 
I found something about the missing files.
1, when i test the following `App.vue` and it will NOT be included in the report
```js
<script setup lang="ts">
import HelloWorld from './components/HelloWorld.vue'
import TheWelcome from './components/TheWelcome.vue'
</script>
```
2, after adding some js codes following and it will be included in the report
```js
<script setup lang="ts">
import HelloWorld from './components/HelloWorld.vue'
import TheWelcome from './components/TheWelcome.vue'
// add some codes
const foo = () => {
  console.log("foo");
}
</script>
```

for case 1, I cann't find the source file (App.vue) in the source map (playwright/.cache/assets/App-********.js.map), so this is why the file not found in the report.
![image](https://github.com/cenfun/monocart-reporter/assets/334577/ceb06ed9-e2cc-40e9-bd05-57fd7d4a1b75)
but for case 2, the App.vue will be found in the source map:
![image](https://github.com/cenfun/monocart-reporter/assets/334577/15df41a2-a368-4fe1-81cf-4b35538a3690)

Basically, the `.vue` file will be compiled into js if it contains any required js. I guess case 1 is nothing js included, the `VerticalNavDesktop.vue` you provided is the same situation, you can try to add some js codes.

Let me know if you have any questions. thanks

**Comment by azamat-sharapov** [comment]
@cenfun this was exactly the issue! thanks for looking into this! 

When I enable `toIstanbul: true` and `lcov: true`, I get following errors in shell console:

```
[MCR] generating report data ...
[MCR] undefined v8toIstanbul.load: ENOENT: no such file or directory, open 'vue.runtime.esm-bundler-dca0544b.js'
[MCR] undefined v8toIstanbul.applyCoverage: Cannot read properties of undefined (reading 'source')
```

But I don't see any problem with generated reports and lcov.info is also there, so this is just FYI.

**Comment by cenfun** [comment]
the error you provided is the `vue.runtime.esm-bundler-dca0544b.js` may fail to load sourcemap.  
good to know it works.
